src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -0,0 +1,319 @@
+        /// </summary>
+        /// <param name="expression">Expression which would be tokenized.</param>
+        /// <param name="binding">Binding which would be used to tokenize the operation and determine the type of each token.</param>
+        /// <param name="comments">Colllection of comment tokens extracted from the given expression.</param>
+        /// <param name="comparer">optional comparer to sort tokens.</param>
+        /// <returns>Ordered or undordered collection of tokens.</returns>
+        internal static ICollection<ITokenTextSpan> Tokenize(string expression, TexlBinding binding, IEnumerable<CommentToken> comments = null, IComparer<ITokenTextSpan> comparer = null)
+        {
+            var tokens = new List<ITokenTextSpan>();
+            if (binding == null)
+            {
+                return tokens;
+            }
+
+            Span span;
+            var complilerGeneratedCallNodes = new HashSet<int>();
+
+            foreach (var firstName in binding.GetFirstNames())
+            {
+                var textSpan = firstName.Node.GetCompleteSpan();
+                span = new Span(textSpan.Min, textSpan.Lim);
+                TokenType type = MapBindKindToTokenType(firstName.Kind);
+
+                // Try to set the canHide flag if the FirstName type is Enum
+                var canHide = CanHideNamespaceOfDottedName(firstName, binding);
+
+                // If the first name can be hidden, Create and Add a token for the dot.
+                if (canHide && expression.Length > span.Lim && (TexlLexer.PunctuatorDot + TexlLexer.PunctuatorBang).IndexOf(expression.Substring(span.Lim, 1), StringComparison.Ordinal) >= 0)
+                {
+                    tokens.Add(new TokenTextSpan(TexlLexer.PunctuatorDot, span.Lim, span.Lim + 1, TokenType.Punctuator, canHide: true));
+                }
+
+                // Add a token for the first name.
+                tokens.Add(new TokenTextSpan(firstName.Name, span.Min, span.Lim, type, canHide));
+            }
+
+            foreach (var dottedName in binding.GetDottedNames())
+            {
+                DType type = binding.GetType(dottedName.Node);
+
+                TokenType dottedNameType = TokenType.DottedNamePart;
+                if (type.IsControl)
+                {
+                    dottedNameType = TokenType.Control;
+                }
+
+                span = dottedName.Node.GetTextSpan();
+                tokens.Add(new TokenTextSpan(dottedName.Node.Right.Name, span.Min + 1, span.Lim, dottedNameType, /* canHide */ false));
+            }
+
+            // String interpolations
+            ExtractTokensFromStringInterpolationNodesList(tokens, binding, complilerGeneratedCallNodes);
+
+            foreach (var call in binding.GetCalls())
+            {
+                // Skip the compiler generated nodes as they don't appear in the actual source code and should not up as a token
+                if (complilerGeneratedCallNodes.Contains(call.Node.Id))
```

Comment:
Binder generates a transient Concatenate call node whose span overlaps with the span $" (start of interpolated string). This node was not skipped in the original version of the whole Tokenization logic and this resulted into two overlapping tokens. Ideally, Concatenate is a compiler generated node and should not appear in the set of tokens which are purely generated out of user typed expression. Since monaco-editor doesn't support overlapping tokens and Concatenate appears before $", $" was dropped and was highlighted as a function. Legacy formula bar suffers from the same problem but since it doesn't use monaco's in built tokenization, it got lucky.